src/plugins/ui_feature/insert_pivot.ts

```diff
@@ -0,0 +1,57 @@
+import { getZoneArea } from "../../helpers/zones";
+import { _t } from "../../translation";
+import { UID } from "../../types";
+import { Command } from "../../types/commands";
+import { UIPlugin } from "../ui_plugin";
+
+export class InsertPivotPlugin extends UIPlugin {
+  static getters = [] as const;
+
+  handle(cmd: Command) {
+    switch (cmd.type) {
+      case "INSERT_NEW_PIVOT":
+        this.insertNewPivot(cmd.pivotId, cmd.newSheetId);
+        break;
+    }
+  }
+
+  insertNewPivot(pivotId: UID, sheetId: UID) {
+    if (getZoneArea(this.getters.getSelectedZone()) === 1) {
+      this.selection.selectTableAroundSelection();
+    }
+    const currentSheetId = this.getters.getActiveSheetId();
+    this.dispatch("ADD_PIVOT", {
+      pivotId,
+      pivot: {
+        dataSet: {
+          zone: this.getters.getSelectedZone(),
+          sheetId: currentSheetId,
+        },
+        columns: [],
+        rows: [],
+        measures: [],
+        name: _t("New pivot"),
+        type: "SPREADSHEET",
+      },
+    });
+
+    const position =
+      this.getters.getSheetIds().findIndex((sheetId) => sheetId === currentSheetId) + 1;
+    const formulaId = this.getters.getPivotFormulaId(pivotId);
+    this.dispatch("CREATE_SHEET", {
+      sheetId,
+      name: _t("Pivot #%s", formulaId),
+      position,
+    });
+    this.dispatch("ACTIVATE_SHEET", {
+      sheetIdFrom: currentSheetId,
+      sheetIdTo: sheetId,
+    });
+    this.dispatch("UPDATE_CELL", {
+      sheetId,
+      col: 0,
+      row: 0,
+      content: "=PIVOT(${formulaId})",
```

Comment:
`formulaId` isn't interpolated. And "=PIVOT(${formulaId})" is a bad expression :upside_down_face: 
a small test to insert a pivot is missing :roll_eyes: 
```suggestion
      content: `=PIVOT(${formulaId})`,
```